src/gui/qgsattributeeditorcontext.h

```diff
@@ -132,6 +133,20 @@ class GUI_EXPORT QgsAttributeEditorContext
      */
     inline QgsMapCanvas *mapCanvas() const { return mMapCanvas; }
 
+    /**
+     * Sets the associated CAD dock widget, \a cadDockWidget, (e.g. to be used in map tools).
+     * \see cadDockWidget()
+     * \since QGIS 3.10
+     */
+    void setCadDockWidget( QgsAdvancedDigitizingDockWidget *cadDockWidget ) { mCadDockWidget = cadDockWidget; }
```

Comment:
Can you tag this as private API?

I'd very much like to replace management of cad data in a separate class next to the widget (which would be the one to put in here, once it's there). The current approach has too much things in the dock widget.

And better move the definition to the cpp file (it's an implementation detail, not much to win by leaving it here, except for longer compile times when it needs to be adjusted).

Reply:
> Can you tag this as private API?

How do you do that? Is there a special doxygen keyword or something?

> I'd very much like to replace management of cad data in a separate class next to the widget (which would be the one to put in here, once it's there). The current approach has too much things in the dock widget.

I agree

> And better move the definition to the cpp file (it's an implementation detail, not much to win by leaving it here, except for longer compile times when it needs to be adjusted).

There is no cpp file at the moment. Do you want me to create one, with just this 2 methods? I'm not sure to understand why it matters.

Reply:
Just add a note that this is unstable API, there is no official tag or similar.

Regarding the cpp, feel free to add one. It doesn't mater that much, it's just about isolation API (headers) from implementation (cpp)

Reply:
OK done